### ISSUE TITLE
Pulsar topic name validation fails despite valid topic name

### ISSUE THREAD

**Issue by MatthiasPetermann** [micronaut-projects/micronaut-pulsar#622]
### Expected Behavior

Pulsar topic names can also contain a dot. Micronaut Pulsar should consider such topics as valid.

### Actual Behaviour

Currently, such topics with dots in their name are rejected by the validation.

### Steps To Reproduce

```
@PulsarSubscription(subscriptionName = "pulsar-jtest-subscription",
        subscriptionType = SubscriptionType.Shared)
public class Consumer {

    private static final Logger LOG = getLogger(Consumer.class);

    @PulsarConsumer(topic = "persistent://public/default/my.topic.name",
            consumerName = "shared-consumer-jtester", schema = MessageSchema.BYTES)
    public void messagePrinter(byte[] message) {
        LOG.info(message.toString());
    }
}
```

### Environment Information

- Linux
- OpenJDK 17

### Version

4.7..2

**Comment by MatthiasPetermann** [comment]
Possibly the fix is as simple as:

```
    /**
     * Regex for validating topic name.
     */
    public static final String TOPIC_NAME_VALIDATOR = "((non-)?persistent://)?((\\w+(-|\\w+)*\\w)|(\\$\\{tenant\\}))/(\\w+(-|\\w+)*\\w)/(\\w+([-.]\\w+)*\\w)";
```

in AbstractPulsarConfiguration.java.

Or is it possible to override this configuration?

**Comment by harissecic** [comment]
> Possibly the fix is as simple as:
> 
> ```
>     /**
>      * Regex for validating topic name.
>      */
>     public static final String TOPIC_NAME_VALIDATOR = "((non-)?persistent://)?((\\w+(-|\\w+)*\\w)|(\\$\\{tenant\\}))/(\\w+(-|\\w+)*\\w)/(\\w+([-.]\\w+)*\\w)";
> ```
> 
> in AbstractPulsarConfiguration.java.
> 
> Or is it possible to override this configuration?

I think you would need to implement `PulsarClientConfiguration` and make it a bean. That bean would have to reimplement most of the stuff in `DefaultPulsarClientConfiguration` if you would like to override behaviour.

Regardless of possibility of override I think if dots are allowed it should be allowed by default and follow Pulsar rules. I don't know when will I have time but it should be as simple as making a pull request on mentioned line of code in your latest comment

**Comment by harissecic** [comment]
Will be released as 2.5.1

**Comment by MatthiasPetermann** [comment]
Thank you very much for addressing the issue so quickly and planning it for an upcoming release. Next time, I’ll be happy to submit a pull request myself!
